File: to-fhir/src/main/java/io/github/dizuker/tofhir/FhirCodings.java
package io.github.dizuker.tofhir;

import org.hl7.fhir.r4.model.Coding;

/**
 * Factory methods for default {@link Coding} templates (system + version, no code/display) used
 * across to-FHIR®. Each method returns a fresh instance, since {@link Coding} is mutable.
 */
public final class FhirCodings {

  /** Returns a fresh LOINC coding template. */
  public static Coding loinc() {
    return new Coding().setSystem(FhirSystems.LOINC).setVersion("2.82");
  }

  /** Returns a fresh SNOMED CT coding template. */
  public static Coding snomed() {
    return new Coding()
        .setSystem(FhirSystems.SNOMED)
        .setVersion("http://snomed.info/sct/900000000000207008/version/20250701");
  }

  /** Returns a fresh OPS coding template. */
  public static Coding ops() {
    return new Coding().setSystem(FhirSystems.OPS).setVersion("2026");
  }

  /** Returns a fresh ATC coding template. */
  public static Coding atc() {
    return new Coding().setSystem(FhirSystems.ATC).setVersion("2026");
  }

  /** Returns a fresh ICD-10-GM coding template. */
  public static Coding icd10gm() {
    return new Coding().setSystem(FhirSystems.ICD10GM).setVersion("2026");
  }

  /** Returns a fresh PZN coding template. */
  public static Coding pzn() {
    return new Coding().setSystem(FhirSystems.PZN);
  }

  private FhirCodings() {
    // Utility class, prevent instantiation
  }
}

File: to-fhir-cs/ToFhir/FhirCodings.cs
using Hl7.Fhir.Model;

namespace ToFhir;

/// <summary>
/// Factory methods for default <see cref="Coding"/> templates (system + version, no code/display)
/// used across to-FHIR®. Each method returns a fresh instance, since <see cref="Coding"/> is
/// mutable.
/// </summary>
public static class FhirCodings
{
    /// <summary>Returns a fresh LOINC coding template.</summary>
    public static Coding Loinc() => new() { System = FhirSystems.Loinc, Version = "2.82" };

    /// <summary>Returns a fresh SNOMED CT coding template.</summary>
    public static Coding Snomed() =>
        new()
        {
            System = FhirSystems.Snomed,
            Version = "http://snomed.info/sct/900000000000207008/version/20250701",
        };

    /// <summary>Returns a fresh OPS coding template.</summary>
    public static Coding Ops() => new() { System = FhirSystems.Ops, Version = "2026" };

    /// <summary>Returns a fresh ATC coding template.</summary>
    public static Coding Atc() => new() { System = FhirSystems.Atc, Version = "2026" };

    /// <summary>Returns a fresh ICD-10-GM coding template.</summary>
    public static Coding Icd10Gm() => new() { System = FhirSystems.Icd10Gm, Version = "2026" };

    /// <summary>Returns a fresh PZN coding template.</summary>
    public static Coding Pzn() => new() { System = FhirSystems.Pzn };
}
